taskcluster/async/asyncclient.py

```diff
@@ -137,14 +137,16 @@ class AsyncBaseClient(BaseClient):
                     superExc=rerr
                 )
 
-            # Handle non 2xx status code and retry if possible
-            try:
-                response.raise_for_status()
-                if response.status == 204:
-                    return None
+            status = response.status
+            if status == 204:
+                return None
 
-            except aiohttp.ClientError as rerr:
-                status = response.status
+            # Handle non 2xx status code and retry if possible
+            if status < 200 or status >= 300:
+                rerr = aiohttp.ClientResponseError(
```

Comment:
I don't think we need to create this exception, at least not here.  Let's just throw the `TaskclusterRestFailure` with these values as own properties.

My concern is that the cause of the exception isn't really in the aiohttp or request library, so it shouldn't have an exception from those packages.  We should still be catching errors from the client library, but independently from status checks.  If you'd like, I can take a look at this in the code.